sample/redo/lib/main.dart

```diff
@@ -140,125 +142,122 @@ class _MyHomePageState extends State<MyHomePage> {
+        (tx, Map<String, dynamic> args) async {
+      int id = args['id'];
+      String text = args['text'];
+      int listId = args['listId'];
+      Iterable<Todo> todos = await todosInListFromTx(tx, listId);
+      int index = todos.isEmpty ? 0 : todos.length;
+
+      final order = getNewOrder(index, todos);
+      await _write(tx, Todo(id, listId, text, false, order));
+    });
+
+    _todoDoneMutator = _replicache.register('todo-done',
+        (tx, Map<String, dynamic> args) async {
+      int id = args['id'];
+      bool complete = args['complete'];
+      final todo = await _read(tx, id);
       if (todo == null) {
         return;
       }
       todo.complete = complete;
-      _write(tx, id, todo);
+      await _write(tx, todo);
     });
-  }
 
-  List<Todo> _activeTodos() {
-    final todos =
-        _allTodos.where((todo) => todo.listId == _selectedListId).toList();
-    todos.sort((t1, t2) => (t1.order - t2.order).sign.toInt());
-    return todos;
-  }
+    _todoRemoveMutator = _replicache.register('todo-remove', (tx, args) async {
+      int id = args['id'];
+      await _del(tx, id);
+    });
 
-  Future<void> _handleReorder(int oldIndex, int newIndex) async {
-    final todos = _activeTodos();
-    int id = todos[oldIndex].id;
-    double order = _getNewOrder(newIndex);
-    // TODO(arv): Should be mutate
-    _replicache.query((tx) async {
-      var todo = await _read(tx, id);
+    _todoReorderMutator =
+        _replicache.register('todo-reorder', (tx, args) async {
+      int listId = args['listId'];
+      int oldIndex = args['oldIndex'];
+      int newIndex = args['newIndex'];
+      List<Todo> todos = (await todosInListFromTx(tx, listId)).toList();
+      int id = todos[oldIndex].id;
+      double order = getNewOrder(newIndex, todos);
+      final todo = await _read(tx, id);
       if (todo == null) {
         return;
       }
       todo.order = order;
-      _write(tx, id, todo);
+      await _write(tx, todo);
     });
   }
 
-  Future<void> _handleRemove(int id) async {
-    // TODO(arv): Should be mutate
-    _replicache.query((tx) async {
-      await _del(tx, id);
-    });
-  }
+  Future<void> _addTodoItem(String text) async {
+    // Only add the task if the user actually entered something
+    if (text.isEmpty) {
+      return;
+    }
 
-  void _selectListId(int listId) {
-    setState(() {
-      _selectedListId = listId;
-    });
+    Random r = Random.secure();
+    int id = r.nextInt(1 << 32);
+    await _todoCreateMutator(
+        {'id': id, 'text': text, 'listId': _selectedListId});
   }
 
-  Future<void> _dropDatabase() async {
-    throw UnimplementedError();
-    // Navigator.pop(context);
-    // var items = await _replicache.scan(prefix: prefix);
-    // for (final ScanItem item in items) {
-    //   await _replicache.del(item.key);
-    // }
-    // await _init();
-  }
+  Future<void> _handleDone(int id, bool complete) =>
+      _todoDoneMutator({'id': id, 'complete': complete});
```

Comment:
Nit, I'd just call these, e.g., `_todoDone` or `_markDone`. We don't need to include the type in the name.

Reply:
I wanted to distinguish the json versions from the user friendly versions.